### PR TITLE
MSC3087(MX-EXT0000): Standardised extensions

### PR DESCRIPTION
**Pull request by erkinalp**
Standardised extensions procedure proposed.

### CODE REVIEW

File: proposals/3087-stdex.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# MSC3087: Standardised extensions
```
Comment (uhoreg) [inline proposals/3087-stdex.md:1]: This is unnecessary.  There is nothing preventing an MSC from defining that its additions are optional, and unstable prefixes have nothing to do with whether or not something is optional or not.

This proposal also lacks sufficient detail.

File: proposals/3087-stdex.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Currently, there is a Matrix specification, which comprises of two main 
+documents and many supplemental addenda. The protocol itself is designed
+to be easily extensible, however, there is currently a gap between spec
+and implementation defined extensions. This defines a new RFC process
+akin to the one we have on MSCs, with a few differences:
```
Comment (turt2live) [inline proposals/3087-stdex.md:7]: This smells strongly of trying to convert Matrix to XMPP style governance where clients have to list which extensions they support so users can pick an appropriate server. We do not want this.

The unstable feature process is deliberate to ensure MSCs/ideas end up in the spec, effectively making any extension system irrelevant. We are painfully aware of some MSCs taking a literal eternity to land, however with every MSC we push into FCP we are getting better at avoiding those scenarios.

This kind of MSC is the stuff we generally like to talk about in #matrix-spec before it is opened, as this is historically not an approach we have been willing to entertain.

**Review by turt2live — CHANGES_REQUESTED** [review]

File: proposals/3087-stdex.md (hunk 1 above)
Comment (uhoreg) [reply proposals/3087-stdex.md:1]: Given that this proposal needs more detail to even be considered, I'm marking this as a draft.

File: proposals/3087-stdex.md (hunk 2 above)
Comment (erkinalp) [reply proposals/3087-stdex.md]: My intent was more like a two-track system like ISO C++ WG and IETF RFCs use, a standards track and an extensionb track, with more and more extensions being merged into the mandatory standards track over time.

Comment (turt2live) [reply proposals/3087-stdex.md]: That is very unclear in this proposal. Considering the MSC process was originally created out of [MSC1779](https://github.com/matrix-org/matrix-doc/pull/1779), I'd expect something of similar detail to MSC1779 for any major change to the proposal process.

**Review by KitsuneRal — CHANGES_REQUESTED** [review]
I basically echo what Hubert and Travis said:
1. The MSC lacks detail to the point it feels like an attempt to put a name on things that don't yet exist; there's no substance in this.
2. It seems (there are really too few things described here to be sure) that this MSC tries to introduce something similar to XMPP extensions; in that case, this MSC should show understanding of the current process and its reasons (your "why" section rather shows lack of that knowledge) and clarify why it effectively reverses the current consolidated specification approach. And then it should at least aspire to have the same level of elaboration and (most importantly) references to other sources as [the respective XEP](https://xmpp.org/extensions/xep-0001.html) has.
3. That's another MSC that entirely disregards the structure recommended in the [MSC proposal template](https://github.com/matrix-org/matrix-doc/blob/master/proposals/0000-proposal-template.md), for no apparent reason.

Due to the reasons above, it's not even a draft for me, it's a no-go.

**Comment by ara4n** [comment]
The reason we have a single spec (containing modules for different feature profiles) is to avoid fragmentation and ensure there is at most one official specified way to implement a given feature in Matrix (for a given version of the spec). Meanwhile folks can use MSCs to go wild and propose their own experiments, and the proven good ones get incorporated into the spec.

I see no value in this proposal over the current process, and instead it risks opening the same mess that XMPP has with XEPs, which Matrix deliberately is set up to avoid.

@mscbot fcp close

File: proposals/3087-stdex.md (hunk 2 above)
Comment (KitsuneRal) [reply proposals/3087-stdex.md]: ISO C++ workgroups do not produce "extensions", and IETF RFCs are not "extensions" either. In both cases that's just a part of the standardisation process, a playground for proposals. Our equivalent to IETF RFCs is exactly MSCs. Hint for the future: provide references to explain your sources of inspiration, and examples to be clearer in what kind of problem and how you're trying to address.

**Comment by Zachinquarantine** [comment]
This seems to me like an XMPP style extension governance rule, which, looking at the above comments, doesn't look too good of an idea for a protocol like Matrix... ¯\_(ツ)_/¯

**Comment by turt2live** [comment]
@Zachinquarantine please use line comments.

**Comment by erkinalp** [comment]
Self-close to pre-empt the SCT's backlog.

**Comment by turt2live** [comment]
@mscbot fcp cancel

**Pull request closed without merging**